libraries/integration/Microsoft.Bot.Builder.ApplicationInsights.Core/TelemetrySaveBodyASPMiddleware.cs

```diff
@@ -21,28 +20,26 @@ public TelemetrySaveBodyASPMiddleware(RequestDelegate next)
             _next = next ?? throw new ArgumentNullException(nameof(next));
         }
 
-        public async Task Invoke(HttpContext context)
+        public async Task Invoke(HttpContext httpContext)
         {
-            if (context.Request.Method == "POST")
-            {
-                var memoryCache = context.RequestServices.GetService(typeof(IMemoryCache)) as IMemoryCache;
+            var request = httpContext?.Request;
+
 
-                if (memoryCache != null)
+            if (request != null && request.Method == "POST")
```

Comment:
Per comments from other PR: You don't want to at least check that it's JSON content before going down this path??? You're affecting _every_ `POST` request to the ASP.NET app with this approach.

Reply:
Good call